src/cursor/cur_file.c

```diff
@@ -227,6 +233,7 @@ __curfile_search_near(WT_CURSOR *cursor, int *exact)
     CURSOR_API_CALL(cursor, session, search_near, CUR2BT(cbt));
     WT_ERR(__cursor_copy_release(cursor));
     WT_ERR(__cursor_checkkey(cursor));
+    F_SET(cbt, WT_CBT_EVICT_REPOSITION);
```

Comment:
search near would still release the page if it calls next or prev?

Reply:
Do you suggest I pass a boolean instead?

Reply:
Yes

Reply:
Updated in [3ef25f6](https://github.com/wiredtiger/wiredtiger/pull/7687/commits/3ef25f6c8058f605c9c77f82ef13baaf7c14ec98)